### PR TITLE
NCP2286: Upgrade to pc-ble-driver 4.1.0

### PR DESCRIPTION
**Pull request by alwa-nordic**
Overhaul in conjunction with upgrading to pc-ble-driver 4.1.0. See commits for details.

### CODE REVIEW

File: src/serialadapter.cpp [hunk 1]
```diff
@@ -28,7 +28,16 @@
  */
 
 #include "serialadapter.h"
-#include "serial_port_enum.h"
+
+#include <nan.h>
+#include <sd_rpc.h>
+
+#include <vector>
+#include <list>
```
Comment (bihanssen) [inline src/serialadapter.cpp]: Is `#include <list>` still needed now that vector collection is being introduced?

File: README.md [hunk 2]
```diff
@@ -1,18 +1,44 @@
...
+
+## Building
+
+If there are no precompiled binaries for your platform, the install script will try to build them. You will need a working C++ compiler and the pc-ble-driver library available by cmake find_package.
+
+The recommended way to get pc-ble-driver is using vcpkg. Clone the vcpkg repo, install nrf-ble-driver for the architecture of your node executable. Then set the environment variable `VCPKG_ROOT` to the full path of the vcpkg repository and it will get automatically picked up when doing `npm install`.
+
+A full example of preparing building on Windows for 64-bit Node:
+
+    $ git clone https://github.com/Microsoft/vcpkg.git
```
Comment (kenr) [inline README.md]: The port is not available on MSFT vcpkg master yet: https://github.com/Microsoft/vcpkg/pull/5938
While waiting for this https://github.com/NordicPlayground/vcpkg has to be used.

File: api/firmwareRegistry.js [hunk 3]
```diff
@@ -42,53 +42,46 @@ const arrayToInt = require('./util/intArrayConv').arrayToInt;
...
 const hexDir = path.join(currentDir, '..', '..', 'pc-ble-driver', 'hex');
-const sdV2Dir = path.join(hexDir, 'sd_api_v2');
-const sdV3Dir = path.join(hexDir, 'sd_api_v3');
 
 const VERSION_INFO_MAGIC = 0x46D8A517;
 const VERSION_INFO_START = 0x20000;
 const VERSION_INFO_LENGTH = 24;
 
+const connectivityVersion = '4.1.0';
+const connectivityVersionMagic = 'ble-connectivity 4.1.0+Mar-21-2019-07-43-44';
```
Comment (kenr) [inline api/firmwareRegistry.js]: Perhaps a better word than Magic can be used? Perhaps the name used for this string in the DFU implementation can be used?

File: README.md (hunk 2 above)
Comment (alwa-nordic) [reply README.md]: Should I state this in the README?

File: api/firmwareRegistry.js (hunk 3 above)
Comment (alwa-nordic) [reply api/firmwareRegistry.js]: I think "magic" is jargon for any string embedded in a binary file for identification purposes. I think this string is called "semVer" in device-setup-js, but I'm nor sure. And it's not consistent with what semver usually means: semantic versioning, which a polar opposite of a date as version. Not sure what to do here. I'll check DFU.

File: CMakeLists.txt [hunk 4]
```diff
@@ -61,11 +53,10 @@ else()
     include(${CMAKE_CURRENT_LIST_DIR}/cmake/gcc.cmake)
 endif()
 
-foreach(SD_API_VER ${SD_API_VERS})
-    string(TOLOWER ${SD_API_VER} SD_API_VER_L)
-    set(CURRENT_TARGET pc-ble-driver-js-${SD_API_VER_L})
+foreach(SD_API_VER "2" "3")
```
[the remark this answers is not in the document]
Comment (alwa-nordic) [reply CMakeLists.txt]: I added a explanatory comment for the version choices. I think this is a good solution, right?

File: api/firmwareRegistry.js (hunk 3 above)
Comment (alwa-nordic) [reply api/firmwareRegistry.js]: Bence suggested "applicationVersionString". "connectivityApplicationVersionString" should be fine, right?

Comment (kenr) [reply api/firmwareRegistry.js]: Sounds fine!

File: README.md (hunk 2 above)
Comment (kenr) [reply README.md]: Yes, and verify if the repo is public.

Comment (alwa-nordic) [reply README.md]: Verified: repo is public

**Comment by bencefr** [comment]
Nice job!

**Pull request merged**